### ISSUE TITLE
Warning for element have annotation `Annotatedfor(...)` and `UnannotatedFor(...)`

### ISSUE THREAD

**Issue by Ao-senXiong** [eisop/checker-framework#876]
Issue a warning if one element(package, class, method ...) have both annotation with one of the value matches.

**Comment by wmdietl** [comment]
Let's widen this issue to cover all defaulting annotations.
For example, this does not produce a warning:

````
import org.checkerframework.framework.qual.DefaultQualifier;
import org.checkerframework.framework.qual.TypeUseLocation;
import org.checkerframework.checker.nullness.qual.Nullable;
import org.checkerframework.checker.nullness.qual.NonNull;

@DefaultQualifier(value = NonNull.class, locations = TypeUseLocation.IMPLICIT_UPPER_BOUND)
@DefaultQualifier(value = Nullable.class, locations = TypeUseLocation.IMPLICIT_UPPER_BOUND)
@DefaultQualifier(value = Nullable.class, locations = TypeUseLocation.IMPLICIT_UPPER_BOUND)
class Demo<T> {
  void foo(T p) {
    p.toString(); // OK?
  }
}
````

`Nullable` is listed twice with the same locations, which should produce a warning.

The same location is used with different qualifiers from the same hierarchy, which should also be a warning.
In my tests `NonNull` always won and the `.toString` never produced an error, even with different orders of the annotations.

**Comment by aosen-xiong** [comment]
#856 had this for `Annotatedfor(...)` and `UnannotatedFor(...)` when introducing UnannotatedFor(...), I am moving this out from JSpecfiy umbrella issue.

**Comment by wmdietl** [comment]
@aosen-xiong I think this incorrectly got listed as closed in #856.
#2066 and #2070 check `DefaultQualifier` conflicts. Can you update the issue with what still needs to be checked and either remove this issue from the closed issues or add any missing tests and actually close it?

**Comment by wmdietl** [comment]
Status after #2066, #2070 and #856, with what is still open.

## Implemented

**Conflicting `@DefaultQualifier`** — two annotations that set the same `TypeUseLocation` in the same qualifier hierarchy to *different* qualifiers are a `conflicting.defaults` error, resolved by source order (the first written wins). #2066; test coverage for the explicit `@DefaultQualifier.List` form added in #2070.

**Conflicting `@AnnotatedFor`/`@UnannotatedFor`** — a `conflicting.annotatedfor` warning when both name the same checker. #856.

**Both on a package declaration**, and a conflicting `@HasQualifierParameter`/`@NoQualifierParameter` pair on a package — #2079 (open). Before it, a package-level conflict was reported only when some class in that package was also compiled, and the qualifier-parameter pair was not checked on a package at all.

## Still open: exact duplicates are not reported

This issue's own example is the case that remains. Of the three annotations there:

```java
@DefaultQualifier(value = NonNull.class, locations = TypeUseLocation.IMPLICIT_UPPER_BOUND)
@DefaultQualifier(value = Nullable.class, locations = TypeUseLocation.IMPLICIT_UPPER_BOUND)
@DefaultQualifier(value = Nullable.class, locations = TypeUseLocation.IMPLICIT_UPPER_BOUND)
class Demo<T> {}
```

the `NonNull`/`Nullable` conflict is now reported, but the fact that `Nullable` is written **twice** is not — and a duplicate with no conflict alongside it is entirely silent:

```java
@DefaultQualifier(value = Nullable.class, locations = TypeUseLocation.FIELD)
@DefaultQualifier(value = Nullable.class, locations = TypeUseLocation.FIELD)
class Dup { Object f = null; }   // compiles with no diagnostic
```

That is deliberate as of #2066: two defaults naming the same qualifier are redundant rather than contradictory, so they were kept legal, and `framework/tests/elementdefault/RedundantWrittenDefaults.java` pins that. Reporting them is therefore a decision to revisit, not just code to write.

One thing to settle first: a duplicate can come from an alias rather than from the user repeating themselves. `@NullMarked` aliases to exactly `@DefaultQualifier(NonNull.class, locations = UPPER_BOUND)`, so

```java
@NullMarked
@DefaultQualifier(value = NonNull.class, locations = TypeUseLocation.UPPER_BOUND)
class AliasDup<T> {}
```

is an exact duplicate, and is silent today. A redundancy diagnostic has to decide whether that counts — arguably yes, since the written annotation adds nothing, but it would fire on code that spells the alias out deliberately.

## Tracked elsewhere

`@DefaultQualifierForUse`/`@NoDefaultQualifierForUse` have no conflict check at all — writing both for the same hierarchy is resolved silently by subtraction in `DefaultQualifierForUseTypeAnnotator`. That, and checking these conflicts on declarations read from bytecode, are #2078.

So this issue stays open for the duplicate/redundancy warning above.
